operator/internal/controller/auto/update_controller.go

```diff
@@ -294,6 +297,18 @@ func (rs *reconcileSession) updateStatus(ctx context.Context, obj *autov1alpha1.
 	meta.SetStatusCondition(&obj.Status.Conditions, *rs.failed)
 	rs.complete.ObservedGeneration = obj.Generation
 	meta.SetStatusCondition(&obj.Status.Conditions, *rs.complete)
+
+	// Update the status message with the latest condition message.
+	// Order of precedence: failed > complete > progressing
+	switch {
+	case rs.failed.Status == metav1.ConditionTrue:
+		obj.Status.Message = rs.failed.Message
+	case rs.complete.Status == metav1.ConditionTrue:
+		obj.Status.Message = rs.complete.Message
+	case rs.progressing.Status == metav1.ConditionTrue:
+		obj.Status.Message = rs.progressing.Message
+	}
+
```

Comment:
Is it really necessary to compute `obj.Status.Message` at this time, in updateStatus?  In `Result()` is the logic that updates the message; isn't `s.obj.Status.Message = res.GetSummary().Message` enough?

Reply:
Second iteration of this removes this logic here.